S4.Rmd

```diff
@@ -52,6 +52,21 @@ Slots and contains can specify the names of S4 classes, S3 classes (if registere
 
 To create a class, you call `setClass()`, supplying these three properties. Lets make this concrete with an example. Here we create two classes: a person with character `name` and numeric `age`, and an `Employee` that inherits slots and methods from `Person`, adding an additional `boss` slot that must be a `Person`.
 
+<!-- FIXME: the underlying model here is of two tables 'Person' and 'Employee' in a database. The 'isa' relationship is an implicit join. You've modelled rows, which means that each instance of the class is a scalar. But R works well on vectors. Effective S4 would model columns, and each instance would contain vectors -- 'People' and 'Employees', though since R uses singular integer(), etc, then 'Person' and 'Employee' but with the understanding that these are of length 0, 1, ...
```

Comment:
On the 'data.frame column', S4 classes generally can't be inserted as columns, even if they have the relevant API. One could use `S4Vectors::DataFrame()` or implement a coercion method to / from data.frame, as in `library(GenomicRanges); GRanges(data.frame(gr=GRanges("1:1-10")))`